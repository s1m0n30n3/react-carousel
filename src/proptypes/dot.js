import PropTypes from "prop-types";

export const dotPropTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
  radius: PropTypes.number,
  margin: PropTypes.number,
  isSelected: PropTypes.bool,
};
