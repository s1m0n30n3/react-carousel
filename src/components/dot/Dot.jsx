import React from "react";

import { classNames } from "helpers";
import { useDotStyles } from "hooks";
import { dotPropTypes } from "proptypes";

export const Dot = ({ size, color, radius, margin, isSelected, ...props }) => {
  const dotStyles = useDotStyles({
    size,
    color,
    radius,
    margin,
  });

  return (
    <button
      {...props}
      className={classNames(["dot--base", isSelected && "dot--selected"])}
      {...dotStyles}
    />
  );
};

Dot.propTypes = dotPropTypes;

Dot.displayName = "Gallery.Dot";
