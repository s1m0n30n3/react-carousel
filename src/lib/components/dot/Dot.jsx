import React from "react";

import { classNames } from "helpers";
import { useDotStyles } from "hooks";
import { dotPropTypes } from "proptypes";

import { base, selected } from "styles/dot.module.css";

export const Dot = ({ size, color, radius, margin, isSelected, ...props }) => {
  const dotStyles = useDotStyles({
    size,
    color,
    radius,
    margin,
  });

  return (
    <button
      {...props}
      className={classNames([base, isSelected && selected])}
      {...dotStyles}
    />
  );
};

Dot.propTypes = dotPropTypes;

Dot.displayName = "Gallery.Dot";
