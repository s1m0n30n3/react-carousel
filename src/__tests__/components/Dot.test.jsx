import React from "react";
import renderer from "react-test-renderer";
import { describe, expect, test } from "vitest";

import { Dot } from "components/dot";

describe("Dot", () => {
  test("renders correctly without props", () => {
    const component = renderer.create(<Dot />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("renders all props", () => {
    const component = renderer.create(
      <Dot size={12} color="#ffffff" radius={6} margin={3} isSelected />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
