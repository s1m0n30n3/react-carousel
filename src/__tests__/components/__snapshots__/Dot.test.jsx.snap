// Vitest Snapshot v1

exports[`Dot > renders all props 1`] = `
<button
  className=""
  style={
    {
      "backgroundColor": "#ffffff",
      "borderRadius": "6px",
      "height": "12px",
      "marginLeft": "3px",
      "marginRight": "3px",
      "width": "12px",
    }
  }
/>
`;

exports[`Dot > renders correctly without props 1`] = `
<button
  className=""
/>
`;
